sources/ecs-gpu-init/cmd/ecs-gpu-init/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+const (
+	// NvidiaGPUInfoFilePath is the file path where GPUs and driver info are saved
+	NvidiaGPUInfoFilePath = "/var/lib/ecs/gpu/nvidia-gpu-info.json"
+)
+
+func main() {
+	if err := generateGPUInfoConfig(); err != nil {
+		panic(err)
+	}
+}
+
+// GenerateGPUInfoConfig generates the configuration used by the ECS agent
+func generateGPUInfoConfig() error {
+	if ret := nvml.Init(); ret != nvml.SUCCESS {
+		return errors.New("Failed to initialize NVML")
+	}
+	defer nvml.Shutdown()
+	version, ret := nvml.SystemGetDriverVersion()
+	if ret != nvml.SUCCESS {
+		return errors.New("Failed to get version")
+	}
+	gpuIDs, err := getGPUDeviceIDs()
+	if err != nil {
+		return err
+	}
+
+	if err := writeGPUInfo(version, gpuIDs); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Can this be simplified? It seems like there'd be a more idiomatic Go construct, e.g.:
```suggestion
	return writeGPUInfo(version, gpuIDs)
```